include/nil/crypto3/zk/commitments/batched_commitment.hpp

```diff
@@ -0,0 +1,246 @@
+                    using commitment_type = CommitmentType;
+                    using field_type = FieldType;
+                };
+
+                // Placeholder commitment scheme works with polynomial_dfs
+                template<typename ParamsType, typename TranscriptType, typename PolynomialType = typename math::polynomial_dfs<typename ParamsType::field_type::value_type>> 
+                class polys_evaluator{
+                public:
+                    using params_type = ParamsType;
+                    using commitment_type = typename ParamsType::commitment_type;
+                    using field_type = typename ParamsType::field_type;
+                    using transcript_type = TranscriptType;
+                    using poly_type = PolynomialType;
+
+                    eval_storage<field_type> _z;
+                    polys_evaluator(){}
+                protected:
+                    std::map<std::size_t, std::vector<poly_type>> _polys;
+                    std::map<std::size_t, bool> _locked; // _locked[batch] is true after it is commited
+                    std::map<std::size_t, std::vector<std::vector<typename field_type::value_type>>> _points;
+                protected:
+                    math::polynomial<typename field_type::value_type> get_V(const std::vector<typename field_type::value_type> &points) const{
+                        math::polynomial<typename field_type::value_type> V = {1};
+                        for( std::size_t xi_index = 0; xi_index < points.size(); xi_index++ ){
+                            V = V * math::polynomial<typename field_type::value_type>({-points[xi_index], 1});
+                        }
+                        return V;
+                    }
+                    math::polynomial<typename field_type::value_type> get_U(std::size_t b_ind, std::size_t poly_ind) const{
+                        auto &points = _points.at(b_ind)[poly_ind];
+                        BOOST_ASSERT(points.size() == this->_z.get_poly_points_number(b_ind, poly_ind));
+                        std::vector<std::pair<typename field_type::value_type,typename field_type::value_type>> U_interpolation_points;
+
+                        U_interpolation_points.resize(points.size());
+                        for(std::size_t k = 0; k < points.size(); k++){
+                            U_interpolation_points[k] = std::make_pair( points[k], this->_z.get(b_ind, poly_ind, k) );
+                        }
+
+                        return math::lagrange_interpolation(U_interpolation_points);
+                    }
+                    std::vector<std::vector<typename field_type::value_type>> get_unique_points_list() const{
+                        std::vector<std::vector<typename field_type::value_type>> unique_points;
+
+                        for(auto const &it:_points){
+                            auto k = it.first;
+                            for( std::size_t i = 0; i < _points.at(k).size(); i++ ){
```

Comment:
I don't want to change this data structure now, because I use indices of unique_points as point identifier in eval_map data structure. Am I right, that unordered set doesn't have a convenient indexation. 

Unique_points size is small. So optimization won't have an important result for us.